### DISCUSSION TITLE
Why, after deploying to Google Cloud Run and sending a request to the endpoint, does it time out?

### DISCUSSION THREAD

**Discussion by atreeon** [serverpod/serverpod#139] [Q&A]
Hi, I've created a docker image locally and have run that successfully locally on localhost.  If I go to the end point locally it executes the server endpoint code and puts data into the db & returns data.  I've now tried to deploy to GCP but my http request to the end point times out.  

I don't get any errors in the logs but I do get the following line which seems to suggest serverpod is running ok
`Server id 0 listening on port 8080`

My http request is exactly the same as my localhost request except for https & the url
https://mypod2y-xyzxyzxyz-ew.a.run.app:8080/example

The database connects to the server.  I validated this by creating an execute command using the postgres package in the server.dart file in the main method using the same credentials as the database section in the config file.

I've set my publicHost name in the config to the url that is served by google cloud run.  I've also changed the publicScheme to https.  I'm leaving the publicPort as 8080.

```
publicHost: mypod2y-xyzxyzxyz-ew.a.run.app
publicPort: 8080
publicScheme: https
```

I added a print statement to the beginning of my endpoint code but it doesn't print out to the log which suggests the code isn't being executed and the endpoint isn't being found.

Any ideas why the endpoint times out?  Is there anything else we need to do when deploying to GCP or any server?

**Comment by vlidholt** [comment]
I obviously do not know your setup specifically, but it's likely that you need to grant access to the ports on the server you are running the serverpod on. It sound like it's running correctly from the output, you just need to get through the firewalls to it.

You can also set a specific configuration for your deployed production server. E.g. if you want to run it on port 80, then start it with:

    dart bin/main.dart --mode production

**Reply by atreeon** [reply]
In the dockerfile that I'm using to build my image I have the following
```
EXPOSE 8080
ENTRYPOINT ["/app/bin/server"]
```

There is also a container port option in Cloud Run on the service
![Screenshot 2022-04-28 at 19 45 50](https://user-images.githubusercontent.com/4139336/165824610-d8429c9e-9f78-4d63-90d9-41a5aaa8d87b.png)

and the port numbers in the config file
```
publicHost: mypod2y-xyzxyzxyz-ew.a.run.app
publicPort: 8080
publicScheme: https

# Main port to run the server on.
port: 8080
# Service port, used for insights from the server while it's running.
servicePort: 8081
```

I've tried various configurations but can't seem to make it happy!  I think this should have exposed the port 8080 so it would accept a get http request with the same port number specified.

**Reply by atreeon** [reply]
so, I've been told that Google Cloud Run doesn't actually allow you to specify a port number.

**Reply by mayurdhurpate** [reply]
@atreeon Could you get `serverpod` to run on Google Cloud Run finally? Cloud Run sounds like a great serverless Docker based backend platform, and would be great if serverpod could be hosted there.

**Reply by atreeon** [reply]
No, I couldn't get it to work in the end.  @vlidholt have you had any luck in getting Serverpod to run on Google Cloud Run?  If I remember rightly there wasn't a way to set the port and Google Cloud Run only allows 8080 (I think).  Basically the avoe didn't really change the port number.

**Reply by atreeon** [reply]
It might be worth converting this to an issue - something like 'get to work on Google Cloud Run'

**Reply by vlidholt** [reply]
There is an issue here: https://github.com/serverpod/serverpod/issues/151
